### PR TITLE
feat(clustering): full sync threshold should be configureable

### PR DESCRIPTION
**Pull request by chronolaw**
### Summary

KAG-5586

### Checklist

- [ ] The Pull Request has tests
- [ ] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

**Review by StarlightIbuki — APPROVED** [review]

**Review by StarlightIbuki — APPROVED** [review]

File: kong/clustering/services/sync/rpc.lua [hunk 1]
```diff
@@ -59,6 +55,9 @@ end
 function _M:init_cp(manager)
   local purge_delay = manager.conf.cluster_data_plane_purge_delay
 
+  -- number of versions behind before a full sync is forced
+  local FULL_SYNC_THRESHOLD = manager.conf.cluster_full_sync_threshold or 512
```
Comment (StarlightIbuki) [inline kong/clustering/services/sync/rpc.lua:59]: Should not be in full capital

**Review by StarlightIbuki — APPROVED** [review]
Not a blocker

File: kong/clustering/services/sync/rpc.lua (hunk 1 above)
Comment (StarlightIbuki) [reply kong/clustering/services/sync/rpc.lua:59]: Also the magic number should be named, something like "DEFAULT_SYNC_THRESHOLD"

Comment (chronolaw) [reply kong/clustering/services/sync/rpc.lua:59]: Added, thanks

File: kong/templates/kong_defaults.lua [hunk 2]
```diff
@@ -43,6 +43,7 @@ cluster_use_proxy = off
 cluster_dp_labels = NONE
 cluster_rpc = off
 cluster_incremental_sync = off
+cluster_full_sync_threshold = 512
```
Comment (ms2008) [inline kong/templates/kong_defaults.lua:46]: Shouldn't a description of this configuration item be added to the `kong.conf.default` file as well?

Comment (chronolaw) [reply kong/templates/kong_defaults.lua:46]: Currently other items of incremental sync are not in kong.conf.default, like `cluster_rpc `, so it did the same thing.

Comment (chobits) [reply kong/templates/kong_defaults.lua:46]: so i think we could resolve this comment, when GA, we will add these options~

**Review by chobits — APPROVED** [review]

**Pull request merged**